### PR TITLE
Delay reading the HttpClient response until the ReceiveData property is accessed

### PR DESCRIPTION
**Pull request by claudiamurialdo**
It allows chunked reading when the ReadChunk method is called.
Issue:103034

### CODE REVIEW

File: dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs [hunk 1]
```diff
@@ -135,6 +140,7 @@ internal byte[] ReceiveData
 		{
 			get
 			{
+				ReadReponseData();
```
Comment (ggallotti) [inline dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs:143]: typo in ReadReponseData, missing "s" ReadResponseData

File: dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs [hunk 2]
```diff
@@ -819,9 +818,9 @@ public void HttpClientExecute(string method, string name)
 					_errDescription = aex.InnerException.Message;
 				else
 					_errDescription = aex.Message;
-				response = new HttpResponseMessage();
-				response.Content = new StringContent(HttpHelper.StatusCodeToTitle(HttpStatusCode.InternalServerError));
-				response.StatusCode = HttpStatusCode.InternalServerError;
+				_chunkedResponse = new HttpResponseMessage();
```
Comment (ggallotti) [inline dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs:821]: Why response was renamed to "chunkedResponse". Shouldn't we keep the original name "response" ? Maybe I'm missing something

File: dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs (hunk 1 above)
Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs]: You're right. The typo has been corrected.

File: dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs (hunk 2 above)
Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs]: You're correct about the naming. The 'chunked' prefix was indeed unnecessary. The former variable 'response' was a local routine variable, whereas the new 'response' variable is now an instance variable. I've simply added the '_' prefix to the previous name 'response' to distinguish it from local variables.

**Review by ggallotti — APPROVED** [review]
🚀

**Pull request merged**